spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/util/RefBuilder.scala

```diff
@@ -0,0 +1,52 @@
+package com.datastax.spark.connector.util
+
+case class ConfigParameter (
+  val name:String,
+  val section: String,
+  val default:Option[Any],
```

Comment:
If we generified `ConfigParameter` to `ConfigParameter[T]`, we could represent both optional and non-optional parameters of any type in a type-safe way. And we could get the default value directly from the ConfigParameter. When generating the docs, you can still easily check if T is an option or not.